Build fails on nightly rustc: vers-vecs 1.10 `unroll` macro clashes with new `unroll` builtin (E0659)

`fm-index 0.3.0` fails to compile on recent **nightly** rustc because of its transitive dependency `vers-vecs 1.10.0`:

```
error[E0659]: `unroll` is ambiguous
  --> .../vers-vecs-1.10.0/src/util/unroll.rs:17:16
   |
17 | pub(crate) use unroll;
   |                ^^^^^^ ambiguous name
   |
   = note: ambiguous because of a name conflict with a builtin attribute
   = note: `unroll` could refer to a built-in attribute
```

rustc: `1.98.0-nightly` (2026-06-23). **Stable still builds fine.**

### Root cause
Recent nightly rustc added an `unroll` *builtin attribute*, which shadows the `unroll` `macro_rules!` in `vers-vecs` (fm-index's wavelet-matrix backend), making its `pub(crate) use unroll;` re-export ambiguous.

### Upstream status
The fix (rename the macro `unroll` → `unroll_n`) has been **merged** into vers-vecs as [Cydhra/vers#43](https://github.com/Cydhra/vers/pull/43) (commit `b9a9ddd`), but it has **not been released to crates.io yet** — `vers-vecs` on crates.io is still `1.10.0`, so `cargo build` of `fm-index` on nightly continues to fail.

### Repro
```sh
cargo new x && cd x && cargo add fm-index && cargo +nightly build
```

### Suggestions
- A nudge to the vers-vecs maintainer for a patch release would unblock this for all `fm-index` users on nightly.
- Once a fixed `vers-vecs` (> 1.10.0) is published, an `fm-index` patch release (or just re-resolving) would fix downstream nightly builds.
- Meanwhile, downstream users can `cargo +stable build`, or pin vers-vecs to the merge commit:
  ```toml
  [patch.crates-io]
  vers-vecs = { git = "https://github.com/Cydhra/vers", rev = "b9a9ddd95c9bed916abc7f000e9c2e6e7c33e729" }
  ```

Thanks for the crate — the run-length `RLFMIndex` is exactly what we needed for a repetitive-text count/locate prototype.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ keywords = ["fm-index", "self-index", "succinct", "search", "vers"]
 categories = ["data-structures", "algorithms"]
 
 [dependencies]
-vers-vecs = { version = "1.5.1", features = ["serde"] }
+vers-vecs = { version = "1.10.1", features = ["serde"] }
 num-traits = "0.2"
 serde = { version = "1.0", features = ["derive"] }
 

